src/org/labkey/test/pages/study/CreateNewVisitPage.java

```diff
@@ -0,0 +1,98 @@
+import org.openqa.selenium.support.ui.Select;
+
+public class CreateNewVisitPage extends LabKeyPage<CreateNewVisitPage.ElementCache>
+{
+    public CreateNewVisitPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public static CreateNewVisitPage beginAt(WebDriverWrapper webDriverWrapper)
+    {
+        return beginAt(webDriverWrapper, webDriverWrapper.getCurrentContainerPath());
+    }
+
+    public static CreateNewVisitPage beginAt(WebDriverWrapper webDriverWrapper, String containerPath)
+    {
+        webDriverWrapper.beginAt(WebTestHelper.buildURL("study", containerPath, "createVisit"));
+        return new CreateNewVisitPage(webDriverWrapper.getDriver());
+    }
+
+    public CreateNewVisitPage setLabel(String label)
+    {
+        elementCache().label.set(label);
+        return this;
+    }
+
+    public CreateNewVisitPage setMinSequence(String range)
+    {
+        elementCache().sequenceNumMin.set(range);
+        return this;
+    }
+
+    public CreateNewVisitPage setMaxSequence(String range)
+    {
+        elementCache().sequenceNumMax.set(range);
+        return this;
+    }
+
+    public CreateNewVisitPage setDescription(String value)
+    {
+        setFormElement(elementCache().description, value);
+        return this;
+    }
+
+    public CreateNewVisitPage setType(String option)
+    {
+        elementCache().type.selectByVisibleText(option);
+        return this;
+    }
+
+    public CreateNewVisitPage setVisitHandling(String value)
+    {
+        elementCache().sequenceNumHandling.selectByVisibleText(value);
+        return this;
+    }
+
+    public CreateNewVisitPage setShowByDefault(boolean value)
+    {
+        elementCache().showByDefault.set(value);
+        return this;
+    }
+
+    public ManageVisitPage clickSave()
+    {
+        clickAndWait(elementCache().save);
+        return new ManageVisitPage(getDriver());
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends LabKeyPage<?>.ElementCache
+    {
+        Input label = new Input(Locator.name("label").findWhenNeeded(this), getDriver());
+        Input sequenceNumMin = new Input(Locator.name("sequenceNumMin").findWhenNeeded(this), getDriver());
+        Input sequenceNumMax = new Input(Locator.name("sequenceNumMax").findWhenNeeded(this), getDriver());
+        WebElement description = Locator.textarea("description").findWhenNeeded(this);
+        Select type = SelectWrapper.Select(Locator.name("typeCode")).findWhenNeeded(this);
+        Select sequenceNumHandling = SelectWrapper.Select(Locator.name("sequenceNumHandling")).findWhenNeeded(this);
+        Checkbox showByDefault = new Checkbox(Locator.name("showByDefault").findWhenNeeded(this));
+        WebElement save = Locator.linkWithText("Save").findWhenNeeded(this);
+        WebElement cancel = Locator.linkWithText("Cancel").findWhenNeeded(this);
```

Comment:
I believe the general practice / style is to mark these as `final`.

Reply:
changed, might be good option if we can update the Page template.